### PR TITLE
RangeHover - fix NPE

### LINKED ISSUE #3245: Exception when hover nonexistent case class
**Describe the bug**

I found a case where HoverProvider throws an exception.It doesn't affect UX much, but I guess it would be good to fix.

**To Reproduce**

1. Write `val a = A(1)` where A is not defined.
2. Put cursor on `1`.
3. See exception in logs:

```
Oct 31, 2021 12:11:29 AM scala.meta.internal.pc.CompilerAccess handleError
SEVERE: null
java.lang.NullPointerException
	at scala.meta.internal.pc.HoverProvider.toHover(HoverProvider.scala:222)
	at scala.meta.internal.pc.HoverProvider.hoverOffset(HoverProvider.scala:128)
	at scala.meta.internal.pc.HoverProvider.$anonfun$hover$1(HoverProvider.scala:24)
	at scala.Option.flatMap(Option.scala:283)
	at scala.meta.internal.pc.HoverProvider.hover(HoverProvider.scala:24)
```

**Expected behavior**

No exceptions.

**Installation:**

- Operating system: Linux
- Editor: Emacs
- Metals version: v0.10.8

**Additional context**

scala 2.13.6, Ammonite

### PR DESCRIPTION
**Pull request by dos65**
Fixes #3245

### CODE REVIEW

**Review by tgodzik — APPROVED** [review]
LGTM!

**Comment by dos65** [comment]
@tgodzik do not merge it, I found a different reason why it happened.

File: metals/src/main/scala/scala/meta/internal/metals/Compilers.scala [hunk 1]
```diff
@@ -363,7 +363,7 @@ class Compilers(
       token: CancelToken
   ): Future[Option[Hover]] = {
     withPCAndAdjustLsp(params) { (pc, pos, adjust) =>
-      pc.hover(CompilerRangeParams.fromPos(pos, token))
+      pc.hover(CompilerRangeParams.offsetOrRange(pos, token))
```
Comment (dos65) [inline metals/src/main/scala/scala/meta/internal/metals/Compilers.scala:366]: The actual reason of issue - hover was always processed as a range

**Review by tgodzik — APPROVED** [review]
LGTM!

**Pull request merged**